### ISSUE TITLE
Refactor code related to extensions and Media

### ISSUE THREAD

**Issue by Flashky** [Flashky/armahaleus-telegram-bot#22]
Class CatServiceImpl has a fragment of code that evaluates the extension from the input URL string, generating a Media from it.

That piece of code might be needed too for the upcoming feature of dog command, therefore, needs a refactoring to avoid code duplicity.

Alternatives:

**Static utility class**

PROS:  

- Easy to code and to use.

CONS: 

- Higher coupling, not depending on abstractions.
- Not mockeable on unit testing.

**Spring based service class**

PROS:

- Singleton due to Spring default bean context.
- Less coupling, based on interfaces.

CONS:

- Need to either add annotations to the class as @Service, making it framework dependant.
- An alternative to the previous is using a configuration class and declaring the bean (no annotation). That would be less coupling, but additional code to setup.

**MapStruct based class**

PROS:

- Basically we are mapping a string URI to a Media object, which sounds reasonable. (?)

CONS:

- Breaks Single Responsability Principle. A mapper responsability should be mapping fields, not evaluating their content.
- The mapping is not straightforward as interface, needs an abstract class and implementation of a custom method.

**Best option**

I guess declaring the class as interface + implementation (no spring annotations) + spring configuration class is the best idea, as it means less coupling and doesn't break the single responsability principle. However, it means additional classes an method to properly test.

**Comment by Flashky** [comment]
Ready for 0.0.6
